### ISSUE TITLE
tensorstore interop and/or required parameters

### ISSUE THREAD

**Issue by joshmoore** [LDeakin/zarrs_tools#8]
I'm likely overseeing some required parameter in terms of the codecs from the input array, but while testing https://github.com/ome/ome2024-ngff-challenge/issues/7#issuecomment-2255436377 with:

```
/Users/jamoore/.cargo/bin/zarrs_reencode \
--chunk-shape 1,5,256,256 \
--shard-shape 4,25,2048,2048 \
--separator / \
--attributes-append '{ "looking forward to": "OME-Zarr v0.5" }' \
4496763-v2.zarr/0 4496763-v3-reencode
```

where `4496763-v2.zarr/0` has the .zarray metadata:

```
{
    "chunks": [
        1,
        1,
        2048,
        2048
    ],
    "compressor": {
        "blocksize": 0,
        "clevel": 5,
        "cname": "lz4",
        "id": "blosc",
        "shuffle": 1
    },
    "dimension_separator": "/",
    "dtype": "<u2",
    "fill_value": 0,
    "filters": null,
    "order": "C",
    "shape": [
        4,
        25,
        2048,
        2048
    ],
    "zarr_format": 2
}
```

and `4496763-v3-reencode/` the zarr.json metadata:

<details><summary>long json</summary>

```
{
  "zarr_format": 3,
  "node_type": "array",
  "shape": [
    4,
    25,
    2048,
    2048
  ],
  "data_type": "uint16",
  "chunk_grid": {
    "name": "regular",
    "configuration": {
      "chunk_shape": [
        4,
        25,
        2048,
        2048
      ]
    }
  },
  "chunk_key_encoding": {
    "name": "default",
    "configuration": {
      "separator": "/"
    }
  },
  "fill_value": 0,
  "codecs": [
    {
      "name": "sharding_indexed",
      "configuration": {
        "chunk_shape": [
          1,
          5,
          256,
          256
        ],
        "codecs": [
          {
            "name": "bytes",
            "configuration": {
              "endian": "little"
            }
          },
          {
            "name": "blosc",
            "configuration": {
              "cname": "lz4",
              "clevel": 5,
              "shuffle": "shuffle",
              "typesize": 2,
              "blocksize": 0
            }
          }
        ],
        "index_codecs": [
          {
            "name": "bytes",
            "configuration": {
              "endian": "little"
            }
          },
          {
            "name": "crc32c"
          }
        ],
        "index_location": "end"
      }
    },
    {
      "name": "blosc",
      "configuration": {
        "cname": "lz4",
        "clevel": 5,
        "shuffle": "shuffle",
        "typesize": 2,
        "blocksize": 0
      }
    }
  ],
  "attributes": {
    "looking forward to": "OME-Zarr v0.5",
    "_zarrs": {
      "description": "This array was created with zarrs",
      "repository": "https://github.com/LDeakin/zarrs",
      "version": "0.15.1"
    }
  }
}
```

</details>

tensorstore complains about reading the reencoded array with:

> ValueError: FAILED_PRECONDITION: Error opening "zarr3" driver: Error reading local file "4496763-v3-reencode/zarr.json": Sharding codec {"configuration":{"chunk_shape":[1,5,256,256],"codecs":[{"configuration":{"endian":"little"},"name":"bytes"},{"configuration":{"blocksize":0,"clevel":5,"cname":"lz4","shuffle":"shuffle","typesize":2},"name":"blosc"}],"index_codecs":[{"configuration":{"endian":"little"},"name":"bytes"},{"name":"crc32c"}]},"name":"sharding_indexed"} is not compatible with subsequent bytes -> bytes codecs [{"configuration":{"blocksize":0,"clevel":5,"cname":"lz4","shuffle":"shuffle","typesize":2},"name":"blosc"}] that apply to the entire shard.  Instead, bytes -> bytes codecs may be specified as inner codecs that apply to each sub-chunk individually. [source locations='tensorstore/driver/zarr3/metadata.cc:487\ntensorstore/driver/zarr3/driver.cc:179\ntensorstore/driver/kvs_backed_chunk_driver.cc:1135\ntensorstore/internal/cache/kvs_backed_cache.h:208\ntensorstore/driver/driver.cc:112'] [tensorstore_spec='{\"context\":{\"cache_pool\":{},\"data_copy_concurrency\":{},\"file_io_concurrency\":{},\"file_io_sync\":true},\"driver\":\"zarr3\",\"kvstore\":{\"driver\":\"file\",\"path\":\"4496763-v3-reencode/\"}}']

**Comment by LDeakin** [comment]
This was a bug, thanks for reporting it. Fixed in 7197bf61e4ca82b59b6399f74a58e10d56328134 and it has been released with v0.5.4.

**Comment by joshmoore** [comment]
Thanks, @LDeakin. Works for me locally.
